### DISCUSSION TITLE
!! What are the rules for configuring placement.json

### DISCUSSION THREAD

**Discussion by ChestnutYYY** [OrchardCMS/OrchardCore#12524] [Q&A]
When i use placement.json in theme project file root, I write some content like below:
![image](https://user-images.githubusercontent.com/13895891/193043034-4aca3781-d562-4e27-a0a9-f80a4792b402.png)

**Scene One**
I define "place":"Part:1|body_3;2" , "place":"Part:1|body_3;1",  "place":"Part:1|body_3;0",  what order can the rendering produce the final html?

**Scene Two**
Can I render newsted out of nested columns, like this:  
![image](https://user-images.githubusercontent.com/13895891/193044304-656c96e5-a841-4e67-9f64-0701dddb17f2.png)

**Scene Three**
How to put a specified Part or Field in the same div element? like this:
![image](https://user-images.githubusercontent.com/13895891/193045303-e0c599d1-1a5e-4d30-b806-6d1f7faef58e.png)

**Scene Four**
If  there are two same PartType on a page,  how to determine one of the sections by configuring placement.json, like this:
![image](https://user-images.githubusercontent.com/13895891/193047080-4a34c57b-f8f8-4937-9179-691da6bcd85a.png)

**### I feel it so diffcult , I really need  your help**
![image](https://user-images.githubusercontent.com/13895891/193046771-4253fe44-c80a-491a-90b9-f184334a040d.png)
![image](https://user-images.githubusercontent.com/13895891/193046774-f8100398-1364-405e-ac53-8dd9179125ea.png)
![image](https://user-images.githubusercontent.com/13895891/193046784-d6e845ef-3e2a-4928-a17d-013ef190f450.png)

**Comment by fuzl-llc** [comment]
I've struggled with placement in Orchard since v1 and have never seen a comprehensive document. If I understood it well enough I'd try to create one, but I have a couple thoughts that might help (at least they helped me).

First, they have a pretty good doc at https://docs.orchardcore.net/en/dev/docs/reference/core/Placement/

That doesn't fully cover it for me and I continue to struggle. For example there seem to be "magic" names like "Content" and "Parts" which it seems are "built in" content zones and I think Content comes before Parts vertically in the page. I don't fully get it so if I'm placing everything anyway, I just put everything in content. I think the number after the colon is the order within that area, so I use Content:0, Content:1, Content2, etc.

After that, you can group in three ways using %, #, and | as described in the doc above to create "cards," tabs, and columns. Here, you can make up your own names e.g. to create a tab with your name of choice.

The other thing that helped me was to realize that each JSON property has an *array* of placements. I kept wondering how I place different things for different parts when the root JSON property is something like TextField and I have text fields in multiple parts. So I wanted to create multiple entries in the root JSON doc like MyPart1_TextField and MyPart2_TextField but really you can only have one root TextField property and then inside that you can have multiple matching cases inside the array. Read the doc above a few times and more and more will become clear.

Lastly, I kept wanting to place individual controls like in my custom part I had 8 different properties and associated editors that I'd defined in my own viewmodel and view. I wanted to place them in different places but then realized you can't really separate them - they have to stay together as a unit. Seems obvious in retrospect but in the confusion of using placement I kept trying to refer to a single field and then remembered all the HTML for those editors is together in a single file. 

I also found it hard to test things since you have to restart the app to see your changes and the build time was slow. In this case you can use the placements feature to do them live in the admin site while the site is running and see results immediately and then transfer them to a placements file when you have it working.

**Reply by ChestnutYYY** [reply]
oh~ thks!!! but i have some questions. after i read the  https://docs.orchardcore.net/en/dev/docs/reference/core/Placement/,  I still don't know how to use the `wrappers`? could you tell me? This will be great help for me, thank you so much again.
![image](https://user-images.githubusercontent.com/13895891/194734762-f9ba69eb-a710-4796-ab8d-7f782c53dee4.png)
